org.eclipse.dartboard/src/org/eclipse/dartboard/preference/DartSDKLocationFieldEditor.java

```diff
@@ -75,9 +78,13 @@ private boolean isValidDartSDK(String location) {
 		} catch (InvalidPathException e) {
 			return false;
 		}
+
+		// See https://github.com/eclipse/dartboard/issues/103
+		List<String> blacklist = Lists.newArrayList("/bin/dart", "/usr/bin/dart");
 		// If the entered file doesn't exist, there is no need to run it
 		// Similarly if the file is a directory it can't be the dart executable
-		if (!Files.exists(path) || Files.isDirectory(path)) {
+		if (!Files.exists(path) || Files.isDirectory(path)
```

Comment:
Files.exists(path)  I think, this check requires earlier?

Reply:
Can you check this code is not enough?

	boolean isWindows = Platform.OS_WIN32.equals(Platform.getOS());

		File file = new File(location + File.separator + "bin" + File.separator + (isWindows ? "dart.exe" : "dart"));
		if (!file.exists() || file.isDirectory()) {
			return false;
		}

		String executablePath = file.getAbsolutePath();// Or .getPath();

Reply:
AFAICS this is the same code we had before, which lets / and /usr pass as valid paths to Dart sdks which they're clearly not. See 
![Peek 2019-07-19 16-21](https://user-images.githubusercontent.com/5540255/61542149-45134e00-aa41-11e9-8da1-26358f87e4d2.gif)